applications/ray/kuberay-tpu-webhook/webhook_main_test.go

```diff
@@ -0,0 +1,595 @@
+		},
+		"multi-host, multi-slice worker group": {
+			// multi-slice, replicaIndex should always be 0-numReplicas for 0-numOfHosts pods
+			numOfHosts:  4,
+			numReplicas: 4,
+		},
+		"multiple worker groups": {
+			// should assign replicaIndex 0-numReplicas and TPU_WORKER_ID 0-numOfHosts
+			// for each respective worker group
+			numOfHosts:         4,
+			numReplicas:        4,
+			additionalGroupStr: "another-worker-group",
+			numOfHosts2:        2,
+			numReplicas2:       3,
+		},
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			sliceToWorkers = make(map[slice][]worker)
+			for i := 0; i < tc.numReplicas; i++ {
+				testPodSlice := slice{instanceName, groupNameStr, namespaceStr, i, tc.numOfHosts}
+				for j := 0; j < int(tc.numOfHosts); j++ {
+					replicaIndex := getReplicaIndex(instanceName, groupNameStr, namespaceStr)
+					assert.Equal(t, i, replicaIndex)
+
+					// add the worker to sliceToWorkers - this would happen in getNextWorkerID
+					testWorker := worker{j, replicaIndex, true}
+					if sliceToWorkers[testPodSlice] == nil {
+						sliceToWorkers[testPodSlice] = []worker{testWorker}
+					} else {
+						sliceToWorkers[testPodSlice] = append(sliceToWorkers[testPodSlice], testWorker)
+					}
+				}
+			}
+			if tc.additionalGroupStr != "" {
+				for i := 0; i < tc.numReplicas2; i++ {
+					testPodSlice := slice{instanceName, tc.additionalGroupStr, namespaceStr, i, tc.numOfHosts2}
+					for j := 0; j < int(tc.numOfHosts2); j++ {
+						replicaIndex := getReplicaIndex(instanceName, tc.additionalGroupStr, namespaceStr)
+						assert.Equal(t, i, replicaIndex)
+
+						// add the worker to sliceToWorkers - this would happen in getNextWorkerID
+						testWorker := worker{j, replicaIndex, true}
+						if sliceToWorkers[testPodSlice] == nil {
+							sliceToWorkers[testPodSlice] = []worker{testWorker}
+						} else {
+							sliceToWorkers[testPodSlice] = append(sliceToWorkers[testPodSlice], testWorker)
+						}
+					}
+				}
+			}
+			assert.Equal(t, tc.numReplicas+tc.numReplicas2, len(sliceToWorkers))
+		})
+	}
+}
+
+func Test_GetNextWorkerID(t *testing.T) {
+	setupTest(t)
+
+	tests := map[string]struct {
+		numOfHosts  int32
+		numReplicas int
+		deletePodID int
+	}{
+		"single-host, single-slice worker group": {
+			// single-slice, replicaIndex should always be 0
+			numOfHosts:  1,
+			numReplicas: 1,
+			deletePodID: -1,
+		},
+		"single-host, multi-slice worker group": {
+			// multi-slice, replicaIndex should always be 0-numReplicas
+			numOfHosts:  1,
+			numReplicas: 4,
+			deletePodID: -1,
+		},
+		"multi-host, single-slice worker group": {
+			// single-slice, replicaIndex should always be 0
+			numOfHosts:  4,
+			numReplicas: 1,
+			deletePodID: -1,
+		},
+		"multi-host, multi-slice worker group": {
+			// multi-slice, replicaIndex should always be 0-numReplicas for 0-numOfHosts pods
+			numOfHosts:  4,
+			numReplicas: 4,
+			deletePodID: -1,
+		},
+		"deleted pod from multi-host group": {
```

Comment:
Do we need a test for multiple worker groups here?

Reply:
Done, added a test case for this to validate that re-created pods will be assigned the correct ID in different worker groups.